### PR TITLE
Dev/kduran/us vbs22 39 vbs22 41 listar usuarios deshabilitar usuarios

### PR DESCRIPTION
**Pull request by kendallduranv**
Solicitude to merge this branch

### CODE REVIEW

File: src/app/services/user.service.ts [hunk 1]
```diff
@@ -33,29 +52,49 @@ export class UserService extends BaseService<IUser> {
...
       }),
       catchError(error => {
+        debugger;
         console.error('Error saving user', error);
         return throwError(error);
       })
     );
   }
+
+  // deleteUserSignal (user: IUser): Observable<any>{
```
Comment (nefi-cenfo) [inline src/app/services/user.service.ts:86]: Is this commented code needed?

File: src/app/pages/auth/login/login.component.scss [hunk 2]
```diff
@@ -44,4 +72,100 @@
...
+//   height: 5px;
+//   background: linear-gradient(to left, #FADCD4 0%, transparent 100%);
+//   width: 100%;
+// }
+
+.footer-logo {
+  width: 2.5vw; 
+  height: 5vh;
+
+  // margin-top: 5px;
```
Comment (nefi-cenfo) [inline src/app/pages/auth/login/login.component.scss:165]: Is this commented code needed?

File: src/app/pages/auth/login/login.component.scss [hunk 3]
```diff
@@ -44,4 +72,100 @@
...
+  // background: transparent;
+  background: #775D68;
+  // background: #FADCD4;
+  background-size: cover;
+  background-position: center;
+  margin-left: 0;
+  // flex-direction: column;
+}
+
+// .footer-line {
```
Comment (nefi-cenfo) [inline src/app/pages/auth/login/login.component.scss:154]: Is this commented code needed?

File: src/app/pages/auth/login/login.component.scss [hunk 4]
```diff
@@ -44,4 +72,100 @@
...
+/* Footer */
+.footer {
+  position: fixed;
+  bottom: 0;
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 10px 20px;
+  // background: transparent;
```
Comment (nefi-cenfo) [inline src/app/pages/auth/login/login.component.scss:145]: Is this commented code needed?

File: src/app/pages/auth/login/login.component.scss [hunk 5]
```diff
@@ -44,4 +72,100 @@
...
+  background: #775D68;
+  // background: #FADCD4;
```
Comment (nefi-cenfo) [inline src/app/pages/auth/login/login.component.scss:147]: Is this commented code needed?

File: src/app/pages/auth/login/login.component.scss [hunk 6]
```diff
@@ -44,4 +72,100 @@
...
+  background-size: cover;
+  background-position: center;
+  margin-left: 0;
+  // flex-direction: column;
```
Comment (nefi-cenfo) [inline src/app/pages/auth/login/login.component.scss:151]: Is this commented code needed?

File: src/app/pages/auth/login/login.component.html [hunk 7]
```diff
@@ -35,32 +32,43 @@ <h4>Sign in to account</h4>
...
+            </div>
           </form>
         </div>
       </div>
     </div>
   </div>
+  <footer class="footer">
+    <div class="footer-line"></div>
+    <img src="../../../../assets/icons/Round_Voiage.webp" alt="Logo" class="footer-logo">
+    <!-- <img src="../../../../assets/icons/Round_Voiage_2.webp" alt="Logo" class="footer-logo"> -->
```
Comment (nefi-cenfo) [inline src/app/pages/auth/login/login.component.html:72]: Is this commented code needed?

File: src/app/interfaces/role.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+
+export enum IRole {
+    admin = "ROLE_ADMIN",
+    user = "ROLE_USER",
+    // superAdmin = 'ROLE_SUPER_ADMIN'
```
Comment (nefi-cenfo) [inline src/app/interfaces/role.ts:5]: Is this commented code needed?

File: src/app/components/user/user-list/user-list.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+        component.deleteUser(userToDelete);
+
+        expect(deleteSpy).toHaveBeenCalledWith(userToDelete);
+        expect(onSuccessSpy).toHaveBeenCalled();
+
+
+    });
+
+    
+    // it('it should show list of users', ()=>{
```
Comment (nefi-cenfo) [inline src/app/components/user/user-list/user-list.component.spec.ts:64]: Is this commented code needed?

File: src/app/components/user/user-list/user-list.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+import { UserFormComponent } from '../user-from/user-form.component';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { By } from '@angular/platform-browser';
+import Swal, { SweetAlertResult } from 'sweetalert2';
+import { NotifyService } from '../../../shared/notify/notify.service';
+import { UserService } from '../../../services/user.service';
+import { signal } from '@angular/core';
+import { IUser } from '../../../interfaces/user';
+
+fdescribe('UserListComponent', () => {
```
Comment (nefi-cenfo) [inline src/app/components/user/user-list/user-list.component.spec.ts:18]: i think it's better remove the 'f'.

File: src/app/components/user/user-list/user-list.component.html [hunk 11]
```diff
@@ -2,32 +2,38 @@
...
+        <th scope="col">Last Update</th>
         <th class="text-center" scope="col">Actions</th>
       </tr>
     </thead>
     <tbody>
-      @for (item of userList; track item.id) {
+      @for (item of userList; track item.user_id) {
       <tr>
-        <th scope="row">{{item.id}}</th>
+        <!-- <th scope="row">{{item.id}}</th> -->
```
Comment (nefi-cenfo) [inline src/app/components/user/user-list/user-list.component.html:20]: Is this commented code needed?

File: src/app/components/app-layout/app-layout.component.scss [hunk 12]
```diff
@@ -0,0 +1,11 @@
+//variables de color 
+$background-color: #6D3C52;
+$sidebar-color:#4B2137;
+$text-color: #ffffff; 
+$hover-background-color: #FADCD4; 
+$active-background-color: #6D3C52; 
+
+.backgorund-color {
+    // background-color: #FADCD4;
```
Comment (nefi-cenfo) [inline src/app/components/app-layout/app-layout.component.scss:9]: Is this commented code needed?

**Review by nefi-cenfo — APPROVED** [review]

**Review by kendallduranv — COMMENTED** [review]
Comments solve

**Review by mur-tz — DISMISSED** [review]

**Review by nefi-cenfo — DISMISSED** [review]

**Review by nefi-cenfo — DISMISSED** [review]

**Review by ichivancrCenfotec — APPROVED** [review]

**Review by mur-tz — DISMISSED** [review]

**Review by nefi-cenfo — APPROVED** [review]

**Review by mur-tz — APPROVED** [review]

**Review by DieS06 — APPROVED** [review]

**Review by ichivancrCenfotec — APPROVED** [review]

**Review by mur-tz — APPROVED** [review]

**Review by nefi-cenfo — APPROVED** [review]

**Review by ichivancrCenfotec — APPROVED** [review]

**Review by DieS06 — APPROVED** [review]

**Review by ichivancrCenfotec — APPROVED** [review]
APROVADO

**Review by nefi-cenfo — APPROVED** [review]

**Review by nefi-cenfo — APPROVED** [review]

**Review by mur-tz — APPROVED** [review]

**Review by mur-tz — APPROVED** [review]
approved

**Pull request closed without merging**